connector/client/client.go

```diff
@@ -9,12 +9,15 @@ import (
 	"net/url"
 	"strconv"
 	"strings"
+	"time"
 
 	"github.com/infracloudio/msbotbuilder-go/connector/auth"
 	"github.com/infracloudio/msbotbuilder-go/schema"
 	"github.com/infracloudio/msbotbuilder-go/schema/customerror"
 )
 
+var cache *jwtCache
```

Comment:
@sahil-lakhwani let's follow the same approach as of JWK cache and get rid of this singleton variable